### PR TITLE
STATUSMSG Support

### PR DESCRIPTION
**Pull request by andymandias**
Shows messages with a STATUSMSG prefix in the corresponding channel, and adds an indicator for those messages to the left of the nickname.  I put the indicator to the left of the nickname so that it couldn't be spoofed by message text.  I believe this addresses #438.

### CODE REVIEW

**Review by casperstorm — APPROVED** [review]
LGTM, @tarkah any feedback?

File: data/src/message.rs [hunk 1]
```diff
@@ -75,12 +75,29 @@ impl From<Encoded> for proto::Message {
...
 }
 
 impl Target {
+    pub fn prefix(&self) -> &Option<char> {
+        match self {
+            Target::Server { .. } => &None,
+            Target::Channel { prefix, .. } => prefix,
+            Target::Query { .. } => &None,
+        }
+    }
```
Comment (tarkah) [inline data/src/message.rs:99]: Nit: Typically you'd return Option<&T>, which you can do by prefix.as_ref()

**Review by tarkah — APPROVED** [review]

**Comment by tarkah** [comment]
LGTM, I haven't tested to see how this looks visually but code looks good  👍

File: data/src/message.rs (hunk 1 above)
Comment (andymandias) [reply data/src/message.rs]: Makes perfect sense!  Thanks for the tip :)

**Pull request merged**